### PR TITLE
Add interface name decoder

### PR DESCRIPTION
**Pull request by aibor**
Add decoder for decoding an network interface index into its name, if the interface is still present.

    ebpf_exporter_inet_frags_queue_insert{interface="eth0"} 2

### CODE REVIEW

**Comment by bobrik** [comment]
Could you add an example that utilizes this?

**Comment by aibor** [comment]
> Could you add an example that utilizes this?

Sure, I have added a simple example using the ifname decoder:

```
# HELP ebpf_exporter_inet_frags_total Netfilter inet fragments per interface
# TYPE ebpf_exporter_inet_frags_total counter
ebpf_exporter_inet_frags_total{interface="eth0",ip_version="6"} 2
```

File: decoder/iface_name.go
Comment (bobrik) [inline decoder/iface_name.go:1]: Let's call it `ifname.go` and `ifname_test.go` to match the decoder name

File: decoder/iface_name_test.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+func TestIfNameDecoder(t *testing.T) {
+	cases := []struct {
+		in    []byte
+		cache map[uint32][]byte
+		out   []byte
+	}{
+		{
+			in:    []byte{0x1, 0x0, 0x0, 0x0},
+			cache: map[uint32][]byte{uint32(1): []byte("lo"), uint32(2734686214): []byte("eth42")},
```
Comment (bobrik) [inline decoder/iface_name_test.go:18]: I think you can introduce a `cache` variable for this and reference it here and below.

File: examples/inet-frags.bpf.c [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+}
+
+SEC("kprobe/inet_frag_queue_insert")
+int BPF_KPROBE(inet_frag_queue_insert, struct inet_frag_queue *q, struct sk_buff *skb, int offset)
+{
+    increment_map_for_skb(&inet_frags_total, skb);
+    return 0;
+}
+
+char __license[] SEC("license") = "GPL";
```
Comment (bobrik) [inline examples/inet-frags.bpf.c:39]: Let's use the following to match other files:

```c
char LICENSE[] SEC("license") = "GPL";
```

File: examples/inet-frags.bpf.c [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+#include <bpf/bpf_tracing.h>
+#include "maps.bpf.h"
+
+struct inet_frags_key_t {
+    u32 ifindex;
+    u8 ip_version;
+};
+
+struct {
+    __uint(type, BPF_MAP_TYPE_HASH);
```
Comment (bobrik) [inline examples/inet-frags.bpf.c:12]: Let's use `BPF_MAP_TYPE_LRU_HASH` with a smaller `max_entries` (64 interfaces is a reasonable assumption).

File: examples/inet-frags.yaml [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+      labels:
+        - name: interface
+          size: 4
+          decoders:
+            - name: ifname
+        - name: ip_version
+          size: 4
+          decoders:
+            - name: uint
+
```
Comment (bobrik) [inline examples/inet-frags.yaml:14]: Let's drop this empty line.

**Review by bobrik — COMMENTED** [review]
There are some failed tests. Please also add some docs to the `README.md`.

Thanks!

**Comment by aibor** [comment]
Thanks for the quick review. I made the requested changes. Also, locally `jsonschema.sh` and `golangci-lint` are happy now.

**Review by bobrik — APPROVED** [review]

**Comment by bobrik** [comment]
Great stuff, thanks!

**Pull request merged**